### ISSUE TITLE
learning rate for pre-training

### ISSUE THREAD

**Issue by sysusicily** [mosaicml/llm-foundry#194]
"After pre-training for a while, the loss did not decrease and there was oscillation. I lowered the learning rate (lr) and restarted the training. However, when I resumed training, the learning rate in the log did not appear to update."

![image](https://github.com/mosaicml/llm-foundry/assets/133207026/33fdccb2-24ed-4e38-bb4f-e59f565d2117)

![image](https://github.com/mosaicml/llm-foundry/assets/133207026/e406a0de-9342-46e7-9bfb-f4fd6359ef7c)

**Comment by vchiley** [comment]
If you restart training from a composer trainer ckpt, the trainer will load EVERYTHING including the LR Schedule that was originally used. See [composer docs for a list of what is loaded](https://docs.mosaicml.com/projects/composer/en/latest/trainer/checkpointing.html#resume-training)

In the config, you can set `load_weights_only: true` to [load only the model weights](https://github.com/mosaicml/llm-foundry/blob/main/scripts/train/train.py#L228) or you can use `load_ignore_keys` to tell the trainer to ignore loading certain aspects of the state dict (ie the lr scheduler).
See [composer docs](https://docs.mosaicml.com/projects/composer/en/latest/api_reference/generated/composer.utils.load_checkpoint.html#load-checkpoint) for more details.
